abz/acousticbrainz.py

```diff
@@ -164,3 +164,185 @@ def process(path):
 def cleanup():
     if os.path.isfile(config.settings["profile_file"]):
         os.unlink(config.settings["profile_file"])
+        
+def generate_dataset(path):
+    """Process a direcotry containing a datase
+    
+    Args:
+        path: the relative or absolute path of the dataser rootdir
+    """
+    if not os.path.exists(path):
+        sys.exit(path + " does not exists!")
+    if not os.path.isdir(path):
+        sys.exit(path + " is not a directory!")
+    datasetdict = scan_dir(os.path.abspath(path))
+    print(datasetdict)
+    # submit dataset using the Dataset API
+    submit_dataset(datasetdict)
+    
+def scan_dir(path):
+    """Scan a directory to create the dataset dictionary basic structure
+    
+    Args:
+        path: the absolute path of the dataser rootdir
+    Returns:
+        datasetdict: a dictionary with the basic structure of the dataset
+    """
+
+    dataset_name = os.path.basename(os.path.normpath(path))
+    # handles the basic dataset dictionary structure
+    datasetdict = _datasetdict_structure(dataset_name)
+    itemuuid = None
+    
+    for dirName, sudDirNames, fileNames in os.walk(path):
+        
+        for subdir in sudDirNames:
+            tmpdict = {}
+            tmpdict['name'] = subdir
+            tmpdict['description'] = ''
+            tmpdict['recordings'] = []
+
+            # scan each subfolder in the dataset root
+            subdir = os.path.join(path, subdir)
+    
+            for dirName, subDirNames, fileNames in os.walk(subdir):
+                ### run the extractor
+                for fileName in fileNames:
+                    # run the extractor
+                    filepath = os.path.join(subdir, fileName)
+                    extractor_output = _dataset_item_extractor(filepath)
+                    # submit dataset item (md5 check)
+                    itemuuid = submit_dataset_item(extractor_output)
+                    if not itemuuid==None:
+                        itemuuid = itemuuid['itemuuid']
+                        tmpdict['recordings'].append(str(itemuuid))
+                        _update_progress(filepath, ":) Get UUID", GREEN)
+                        print()
+                    
+                    os.unlink(extractor_output)
+                    
+                if len(tmpdict['recordings'])>0:
+                    datasetdict['classes'].append(tmpdict)
+    
+    #print(datasetdict)
+    return datasetdict
+
+def _dataset_item_extractor(filepath):
+    """Extract lowlevel data from a dataset item
+    
+    Args:
+        filepath:   the fill path of the dataset item to process
+    Returns:
+        tmpname:    the full path of the generated lowlevel-data json file
+    """
+    fd, tmpname = tempfile.mkstemp(suffix='.json')
+    os.close(fd)
+    os.unlink(tmpname)
+    # avoid using default config file for datasets items MBID is not required
+    config.settings["profile_file"] = ""
```

Comment:
We still need the profile file here. Some settings are important, such as the json indent and added data (`mergeValues`) - https://github.com/MTG/acousticbrainz-client/blob/master/abz/config.py#L62
I would prefer to only have this function once, and give it a parameter for which profile file to use. You can generate both profile files at startup.